### PR TITLE
Add recipe for asyncio-helpers-0.2.1

### PR DESCRIPTION
**Pull request by cfobel**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/asyncio-helpers/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/sci-bots/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: ee1c3b674fb58b3362c886290601007f08ef11db4119688d80771ec0cc0ec696
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  skip: true  # [py>36]
```
Comment (chrisburr) [inline recipes/asyncio-helpers/meta.yaml]: Could this be made `noarch`?
```suggestion
  noarch: python
```

File: recipes/asyncio-helpers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - logging-helpers
+    - python >=2.7,<=3.6
+    - trollius  # [py<=36]
```
Comment (chrisburr) [inline recipes/asyncio-helpers/meta.yaml]: Required for `noarch`:
```suggestion
    - trollius
```

File: recipes/asyncio-helpers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  sha256: ee1c3b674fb58b3362c886290601007f08ef11db4119688d80771ec0cc0ec696
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+  skip: true  # [py>36]
+
+requirements:
+  host:
+    - python
```
Comment (chrisburr) [inline recipes/asyncio-helpers/meta.yaml]: Required for `noarch`:
```suggestion
    - python >=2.7,<=3.6
```

**Comment by cfobel** [comment]
Hi Chris,
Thank you for your suggestion. Ideally I would like to make this
package noarch:
python, but there are two issues preventing this:

   1. It currently only supports trollius, which is only available for
   Python <=3.6. There is an existing issue (see sci-bots/asyncio-helpers#2
   <https://github.com/sci-bots/asyncio-helpers/issues/2>) to add support
   for Python 3.4+ asyncio.
   2. Even once sci-bots/asyncio-helpers#2
   <https://github.com/sci-bots/asyncio-helpers/issues/2> is resolved,
   selectors (e.g., # [py<34]) will be required to only require trollius
   where necessary.

So, until sci-bots/asyncio-helpers#2
<https://github.com/sci-bots/asyncio-helpers/issues/2> is implemented,
selectors are required to ensure Python <=3.6, since trollius not available
for Python >3.6.

Additionally, since selectors cannot be used with noarch packages
<https://github.com/conda-forge/conda-smithy/issues/578>, the recipe cannot
be made noarch.

Please let me know if you have any questions or suggestions on other ways
to work around these issues.

Thanks,
Christian

**Comment by chrisburr** [comment]
Ah okay, I suggested `noarch` as the recipe also has `skip: true  # [py<36]` so it could be a `noarch` package that doesn't support Python 3.7.

As you know that the next release will require selectors to support Python 3.7 it makes sense to leave it like this. In that case you should replace `python >=2.7,<=3.6` with `python` as there will be separate builds for each platform and Python version and the version constraint comes from the `skip`.

**Comment by cfobel** [comment]
> Ah okay, I suggested `noarch` as the recipe also has `skip: true # [py<36]` so it could be a `noarch` package that doesn't support Python 3.7.

@chrisburr thanks for the tip on how to mark the recipe as `noarch` while restricting the Python version. 
 In 1a6a2d46bcd93369cfa1d61e1b1343b569a66e16, I took your recommended approach:

 - add `noarch: python` to `build`
 - set `python >=2.7,<3.7` under `run` requirements

> As you know that the next release will require selectors to support Python 3.7 it makes sense to leave it like this. In that case you should replace `python >=2.7,<=3.6` with `python` as there will be separate builds for each platform and Python version and the version constraint comes from the `skip`.

Once I add support to the library for Python 3.7+, I will:

 - remove `noarch: python` from `build`
 - add `# [py<37]` selector to `trollius` run requirement
 - replace `python >=2.7,<3.7` run requirement with `python`

**Comment by cfobel** [comment]
Thanks @chrisburr !

**Pull request merged**